breathe/project.py

```diff
@@ -172,9 +172,12 @@ def update(
             build_dir
             ):
 
+        # Provide reasonable defaults for domain_by_extension mapping. Can be overridden by users.
+        default_domain_by_extension = {'py': 'py'}
```

Comment:
I think it is fair to default python files to the python domain, good call. However, maybe there is an argument for having this as a default value for the breathe_domain_by_extension set here? I realise that means that if a user sets a c++ mapping then they might unintentionally override the python mapping but I think I'm ok with that given that we'd be using the standard Sphinx default option mechanism.